### ISSUE TITLE
Pygments 2.12 breaks insiders build

### ISSUE THREAD

**Issue by saltydk** [squidfunk/mkdocs-material#3838]
### Contribution guidelines

- [X] I've read the [contribution guidelines](https://github.com/squidfunk/mkdocs-material/blob/master/CONTRIBUTING.md) and wholeheartedly agree

### I've found a bug and checked that ...

- [X] ... the problem doesn't occur with the `mkdocs` or `readthedocs` themes
- [X] ... the problem persists when all overrides are removed, i.e. `custom_dir`, `extra_javascript` and `extra_css`
- [X] ... the documentation does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem

### Description

Recent release of pygments breaks mkdocs material.

When I pin the version to 2.10 (haven't tested 2.11 since I went straight to one I knew that worked) it works like normal.

### Expected behaviour

Building completes.

### Actual behaviour

```
ERROR    -  Error reading page 'apps/autoscan.md': format() argument must be a formatter instance, not a class
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.10.4/x64/lib/python3.10/site-packages/pygments/__init__.py", line 61, in format
    formatter.format(tokens, realoutfile)
  File "/opt/hostedtoolcache/Python/3.10.4/x64/lib/python3.10/site-packages/pygments/formatter.py", line 94, in format
    return self.format_unencoded(tokensource, outfile)
  File "/opt/hostedtoolcache/Python/3.10.4/x64/lib/python3.10/site-packages/pygments/formatters/html.py", line 981, in format_unencoded
    source = self.wrap(source)
TypeError: BlockHtmlFormatter.wrap() missing 1 required positional argument: 'outfile'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.10.4/x64/bin/mkdocs", line 8, in <module>
    sys.exit(cli())
  File "/opt/hostedtoolcache/Python/3.10.4/x64/lib/python3.10/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/opt/hostedtoolcache/Python/3.10.4/x64/lib/python3.10/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/opt/hostedtoolcache/Python/3.10.4/x64/lib/python3.10/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/opt/hostedtoolcache/Python/3.10.4/x64/lib/python3.10/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/opt/hostedtoolcache/Python/3.10.4/x64/lib/python3.10/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/opt/hostedtoolcache/Python/3.10.4/x64/lib/python3.10/site-packages/mkdocs/__main__.py", line 215, in gh_deploy_command
    build.build(cfg, dirty=not clean)
  File "/opt/hostedtoolcache/Python/3.10.4/x64/lib/python3.10/site-packages/mkdocs/commands/build.py", line 292, in build
    _populate_page(file.page, config, files, dirty)
  File "/opt/hostedtoolcache/Python/3.10.4/x64/lib/python3.10/site-packages/mkdocs/commands/build.py", line 174, in _populate_page
    page.render(config, files)
  File "/opt/hostedtoolcache/Python/3.10.4/x64/lib/python3.10/site-packages/mkdocs/structure/pages.py", line 175, in render
    self.content = md.convert(self.markdown)
  File "/opt/hostedtoolcache/Python/3.10.4/x64/lib/python3.10/site-packages/markdown/core.py", line 268, in convert
    newRoot = treeprocessor.run(root)
  File "/opt/hostedtoolcache/Python/3.10.4/x64/lib/python3.10/site-packages/pymdownx/highlight.py", line 491, in run
    code.highlight(
  File "/opt/hostedtoolcache/Python/3.10.4/x64/lib/python3.10/site-packages/pymdownx/highlight.py", line 392, in highlight
    code = highlight(src, lexer, formatter)
  File "/opt/hostedtoolcache/Python/3.10.4/x64/lib/python3.10/site-packages/pygments/__init__.py", line 82, in highlight
    return format(lex(code, lexer), formatter, outfile)
  File "/opt/hostedtoolcache/Python/3.10.4/x64/lib/python3.10/site-packages/pygments/__init__.py", line 69, in format
    raise TypeError('format() argument must be a formatter instance, '
TypeError: format() argument must be a formatter instance, not a class```

### Steps to reproduce

Build with pygments 2.12.0 installed

### Package versions

https://github.com/saltyorg/docs/runs/6148823777?check_suite_focus=true

### Configuration

```yaml
Doesn't seem relevant.
```

### System information

Ubuntu 20.04 using Github Actions.

**Comment by squidfunk** [comment]
Already fixed in 85b087ac9. Pinned to <2.12.

**Comment by saltydk** [comment]
Thanks :)
